cmd/kubeadm/app/util/patches/patches.go

```diff
@@ -366,3 +366,37 @@ return_path_error:
 
 	return patchSets, patchFiles, ignoredFiles, nil
 }
+
+// GetSinglePatchFromFile reads a single patch file and returns the patch data and its type.
+func GetSinglePatchFromFile(patchFilePath string, output io.Writer) ([]byte, error) {
```

Comment:
instead of adding these two new functions in the patches.go file, let's use the existing API
it seems simpler like you did but it adds more extra code.

do the following:
- make createPatchSet public -> CreatePatchSet
- create a new function

```
func NewPatchManager(patchSets []*patchSet, knownTargets []string, output io.Writer) *PatchManager {
	return &PatchManager{
		patchSets:    patchSets,
		knownTargets: knownTargets,
		output:       output,
}
```

in phases/kubelet/config.go

add a new function that does the following:
- it loads the instance-coffig patch file as bytes
- it calls CreatePatchSet to create a patchset from it target should be set to "kubeletconfiguration"
- calls NewPatchManager - knownTargets should have "kubletconfiguration"
- create a new PatchTarget and call ApplyPatchesToTarget

similar to this function
https://github.com/kubernetes/kubernetes/blob/721d66780b1797b49c3d6119a92533b6ff58eec2/cmd/kubeadm/app/phases/kubelet/config.go#L206-L207